File: okj/szoftverfejleszto/operatorok/java/Kifejezes.java
public class Kifejezes {

    public final int elsoOperandus;
    public final String operator;
    public final int masodikOperandus;
    public final String teljesKifejezes;

    public Kifejezes(String line) {
        var split = line.split(" ");

        this.elsoOperandus = Integer.parseInt(split[0]);
        this.operator = split[1];
        this.masodikOperandus = Integer.parseInt(split[2]);
        this.teljesKifejezes = line;
    }

    // A Math.exact fv-ek exception-t dobnak overflow esetén
    public String kiertekel() {
        try {
            return String.valueOf(
                    switch(operator) {
                        case "+" -> Math.addExact(elsoOperandus, masodikOperandus);
                        case "-" -> Math.subtractExact(elsoOperandus, masodikOperandus);
                        case "*" -> Math.multiplyExact(elsoOperandus, masodikOperandus);
                        case "/" -> ((double) elsoOperandus) / ((double) masodikOperandus);
                        case "div" -> elsoOperandus / masodikOperandus;
                        case "mod" -> elsoOperandus % masodikOperandus;
                        default -> "Hibás operátor!";
                    });
        }catch (ArithmeticException e) {
            return "Egyéb hiba!";
        }
    }
}

File: okj/szoftverfejleszto/operatorok/c#/Kifejezes.cs
using System;

public class Kifejezes {

    public readonly int ElsoOperandus;
    public readonly string Operator;
    public readonly int MasodikOperandus;
    public readonly string TeljesKifejezes;

    public Kifejezes(string line) {
        var split = line.Split(' ');

        this.ElsoOperandus = int.Parse(split[0]);
        this.Operator = split[1];
        this.MasodikOperandus = int.Parse(split[2]);
        this.TeljesKifejezes = line;
    }

    public string kiertekel() {
        checked {
            try {
                switch(Operator) {
                    case "+": return (ElsoOperandus + MasodikOperandus).ToString();
                    case "-": return (ElsoOperandus - MasodikOperandus).ToString();
                    case "*": return (ElsoOperandus * MasodikOperandus).ToString();
                    case "/": return ((double) ElsoOperandus / (double) MasodikOperandus).ToString();
                    case "div": return (ElsoOperandus / MasodikOperandus).ToString();
                    case "mod": return (ElsoOperandus % MasodikOperandus).ToString();
                    default: return "Hibás operátor!";
                }
            }catch (ArithmeticException _) {
                return "Egyéb hiba!";
            }
        }
    }
}